src/Dfc.CourseDirectory.Core/DataStore/Sql/QueryHandlers/GetProviderDashboardCountsHandler.cs

```diff
@@ -1,39 +1,75 @@
-﻿using System.Data.SqlClient;
+﻿using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
 using System.Threading.Tasks;
 using Dapper;
 using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
+using Dfc.CourseDirectory.Core.Models;
 
 namespace Dfc.CourseDirectory.Core.DataStore.Sql.QueryHandlers
 {
     public class GetProviderDashboardCountsHandler :
-        ISqlQueryHandler<GetProviderDashboardCounts, (int CourseRunCount, int ApprenticeshipCount, int VenueCount)>
+        ISqlQueryHandler<GetProviderDashboardCounts, (IReadOnlyDictionary<CourseStatus, int> CourseRunCounts, IReadOnlyDictionary<ApprenticeshipStatus, int> ApprenticeshipCounts, int VenueCount, int PastStartDateCourseRunCount, int BulkUploadCoursesErrorCount, int BulkUploadCourseRunsErrorCount, int ApprenticeshipsBulkUploadErrorCount)>
     {
-        public async Task<(int CourseRunCount, int ApprenticeshipCount, int VenueCount)> Execute(
+        public async Task<(IReadOnlyDictionary<CourseStatus, int> CourseRunCounts, IReadOnlyDictionary<ApprenticeshipStatus, int> ApprenticeshipCounts, int VenueCount, int PastStartDateCourseRunCount, int BulkUploadCoursesErrorCount, int BulkUploadCourseRunsErrorCount, int ApprenticeshipsBulkUploadErrorCount)> Execute(
             SqlTransaction transaction,
             GetProviderDashboardCounts query)
         {
-            var sql = @"
+            var sql = @$"
 DECLARE @ProviderUkprn INT
 
-SELECT TOP 1 @ProviderUkprn = Ukprn FROM Pttcd.Providers WHERE ProviderId = @ProviderId
+SELECT TOP 1 @ProviderUkprn = Ukprn FROM Pttcd.Providers WHERE ProviderId = @{nameof(query.ProviderId)}
 
-SELECT COUNT(*) FROM Pttcd.Courses c
-JOIN Pttcd.CourseRuns cr ON c.CourseId = cr.CourseId
-WHERE c.ProviderUkprn = @ProviderUkprn AND cr.CourseRunStatus = 1
+SELECT		cr.CourseRunStatus, COUNT(*) as Count
```

Comment:
Nit: indentation is off here

Reply:
Note to self, beware copy and past from SSMS!